js/demo_policy/main.ts

```diff
@@ -0,0 +1,202 @@
+import { ManifestDigest } from '../kubewarden/host_capabilities/oci/manifest_digest/manifest_digest';
+
+declare function policyAction(): string;
+
+interface PolicySettings {
+  ignoredNamespaces?: string[];
+  testScenario?: string;
+}
+
+class PolicySettings {
+  constructor(ignoredNamespaces: string[] = [], testScenario?: string) {
+    this.ignoredNamespaces = ignoredNamespaces;
+    this.testScenario = testScenario;
+  }
+}
+
+/**
+ * Handles OCI manifest digest lookup success scenario
+ */
+function handleOciManifestDigestSuccess(): Validation.ValidationResponse {
+  const image = 'ghcr.io/kubewarden/kubectl:v1.31.0';
+  let digest: string | null = null;
+  try {
+    digest = ManifestDigest.getOCIManifestDigest(image);
+  } catch (e) {
+    console.error('OCI manifest digest lookup failed:', e);
+  }
+  return new Validation.ValidationResponse(
+    !!digest,
+    digest ? undefined : 'Failed to retrieve OCI manifest digest',
+    undefined,
+    undefined,
+    { digest: digest || '' }
+  );
+}
+
+/**
+ * Handles OCI manifest digest lookup failure scenario
+ */
+function handleOciManifestDigestFailure(): Validation.ValidationResponse {
+  const image = 'registry.testing.lan/nonexistent-image:1.0.0';
+  let digest: string | null = null;
+  try {
+    digest = ManifestDigest.getOCIManifestDigest(image);
+  } catch (e) {
+    console.error('OCI manifest digest lookup failed:', e);
+    return new Validation.ValidationResponse(
+      false,
+      `OCI manifest digest lookup failed: ${e}`,
+      undefined,
+      undefined,
+      { digest: '' }
+    );
+  }
+  // If digest is empty, treat as failure
+  return new Validation.ValidationResponse(
+    false,
+    'Unexpectedly retrieved OCI manifest digest',
+    undefined,
+    undefined,
+    { digest: digest }
+  );
+}
+
+/**
+ * Handles DNS lookup success scenario
+ */
+function handleDnsLookupSuccess(): Validation.ValidationResponse {
+  let ips: string[] = [];
+  try {
+    ips = Network.dnsLookup('google.com').ips || [];
+  } catch (e) {
+    console.error('DNS lookup failed:', e);
+  }
+  return new Validation.ValidationResponse(
+    !!ips && ips.length > 0,
+    ips && ips.length > 0 ? undefined : 'Failed to retrieve DNS lookup IPs',
+    undefined,
+    undefined,
+    { ips: ips.join(', ') || '' }
+  );
+}
+
+/**
+ * Handles DNS lookup failure scenario
+ */
+function handleDnsLookupFailure(): Validation.ValidationResponse {
+  let ips: string[] = [];
+  try {
+    ips = Network.dnsLookup('invalid.nonexistent.tld').ips || [];
+    if (ips.length > 0) {
+      return new Validation.ValidationResponse(
+        false,
+        'Unexpectedly retrieved DNS lookup IPs',
+        undefined,
+        undefined,
+        { ips: ips.join(', ') }
+      );
+    } else {
+      return new Validation.ValidationResponse(
+        false,
+        'DNS lookup returned no results',
+        undefined,
+        undefined,
+        { ips: '' }
+      );
+    }
```

Comment:
I don't genuinely remember what happens in this case when the FQDN cannot be found. Are we going to raise an error or are we returning an empty list? Please do some experiment or look at the code. Then we can decide about how to rewrite this code, I don't think we have to leave the `if/else`

I would move the check on the ips outside of the try/catch